server/plugin.go

```diff
@@ -557,6 +570,15 @@ func (p *Plugin) syncUsers() {
 				break
 			}
 		}
+
+		if (configuration.AutomaticallyPromoteSyntheticUsers && (mmUser.AuthService != configuration.SyntheticUserAuthService || mmUser.AuthData != &configuration.SyntheticUserAuthData)) && mmUser.RemoteId != nil {
```

Comment:
How can user's auth data be compared to `configuration.SyntheticUserAuthData`? The configuration contains field name while the user's auth data contains value. Also, why haven't we put this logic in the `if` condition at line no. 455.

Reply:
I ended up downloading the .webm video -- couldn't watch inline -- and realized there's a part of this I wasn't expecting: changing the configuration sets all those synthetic users from `Email` to `ldap` or `saml` immediately: even before the login. This surprised me, as I thought this only happened on login, but I guess makes sense in hindsight if that's what's required to let them login in the first place.

Can you confirm?

Reply:
Yes, you are correct. We update the `Auth Sevice` of a user on the basis of the configurations, as it is required for them to login and get auto-promoted.